cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -185,6 +185,95 @@ The diagram for this code would be:
+
+    @Override
+    public void configure() {
+      setName("ConditionalWorkflow");
+      setDescription("Demonstration of conditional execution of a Workflow");
+      
+      addMapReduce("JoinWithCatalogMR");
+      
+      condition(new MyPredicate())
+        .addMapReduce("BuildProductProfileMR")
+      .otherwise()
+        .addMapReduce("BuildUserProfileMR")
+      .end();
+      
+      addAction(new UploadProfilesCA());
+    }
+  }
+
+where ``MyPredicate`` is a public class which implements the ``Predicate`` interface as::
+
+  public static class MyPredicate implements Predicate<WorkflowContext> {
+
+    @Override
+    public boolean apply(@Nullable WorkflowContext input) {
+      if (input != null && input.containsKey("BuildProductProfile")) {
+        Map<String, Long> customCounters = input.getToken().getMapReduceCounters().get("MyCustomCounters");
+        if (customCounters.get("BuildProductProfile") > 0) {
+          return true;
+        }
+      }
+      return false;
+    }
+  }
+  
+In the ``JoinWithCatalogMR`` MapReduce, it could have in its Mapper class code that 
+governs which condition to follow::
+
+  public static final class JoinWithCatalogMR extends AbstractMapReduce {
+
+    @Override
+    public void configure() {
+      setName("JoinWithCatalogMR");
+      setDescription("MapReduce program to demonstrate a Conditional Workflow");
+    }
+
+    @Override
+    public void beforeSubmit(MapReduceContext context) throws Exception {
+      Job job = context.getHadoopJob();
+      job.setMapperClass(MyVerifier.class);
+      String inputPath = context.getRuntimeArguments().get("inputPath");
+      String outputPath = context.getRuntimeArguments().get("outputPath");
+      FileInputFormat.addInputPath(job, new Path(inputPath));
+      FileOutputFormat.setOutputPath(job, new Path(outputPath));
+    }
+  }
+
+  public static class MyVerifier extends Mapper<LongWritable, Text, Text, NullWritable> {
+    public void map(LongWritable key, Text value, Context context)
+      throws IOException, InterruptedException {
+      if (value != null and value.toString().equals("BuildProductProfile")) {
+        context.getCounter("MyCustomCounters", "BuildProductProfile").setValue(1L);
+      } else {
+        context.getCounter("MyCustomCounters", "BuildProductProfile").setValue(0);
+      }
+    }
+  }
+
+In this case, if the predicate finds that the a ``MapReduceCounter`` *BuildProductProfile*
+is greater than zero, the logic will follow the path of *BuildProductProfileMR*;
+otherwise, the other path will be taken. The diagram for this code would be:
+
+.. image:: /_images/conditional-workflow.png
+   :width: 8in
+   :align: center
+
+For this 3.0 release, CDAP only supports predicate conditions based on counters from your
+MapReduce programs. Because of this, condition nodes can currently only follow MapReduce
+nodes that make use of one or more counters. In a later version, we will expose many more
```

Comment:
I mean condition nodes as a CDAP feature will still work right (in the sense that the condition will be executed, will most likely always make it to one branch of the condition node), it probably may not make sense for applications to use them?